File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter05/RemotingInterfaces/CustomerClient/CustClient.cs
using System;
using System.Runtime.Remoting;
using CustomerLibrary;

namespace CustomerClient
{
   class CustomerMain
   {
      static void Main(string[] args)
      {
         RemotingConfiguration.Configure("CustomerClient.exe.config");
         
         // Build a proxy to the CustomerService object
         ICustomerService custSvc;
         custSvc = (ICustomerService)Activator.GetObject(
                      typeof(ICustomerService),
                      "http://localhost:13101/customer.soap"
                   );

         // Call ICustomerService.Test to confirm remoting settings
         Console.WriteLine(custSvc.Test());

         // Get a proxy to a remote Customer object
         ICustomer cust = custSvc.CreateCustomer("Homer");
         Console.WriteLine(cust.SayHello());

         Console.ReadLine();
      }
   }
}

File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter05/RemotingFactory/CustomerClient/CustClient.vb
Imports System.Runtime.Remoting
Imports CustomerInterfaces

Module CustomerMain

   Sub Main()
        
      'Use the CustomerFactory to get an ICustomerService object         
      Dim custSvc As ICustomerService = CustomerFactory.CreateCustomerService()
      
      'Execute the simple Test method
      Console.WriteLine(custSvc.Test())

      'Get a proxy to a remote Customer object
      Dim cust As ICustomer = custSvc.CreateCustomer("Homer")
      Console.WriteLine(cust.SayHello())
      
      Console.ReadLine()
   End Sub
End Module